src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItemSource.cs

```diff
@@ -32,15 +33,18 @@ internal class SourceGeneratedFileItemSource : Shell.IAttachedCollectionSource,
         private readonly BulkObservableCollectionWithInit<BaseItem> _items = new();
 
         /// <summary>
-        /// Gate to guard mutation of <see cref="_cancellationTokenSource"/> and <see cref="_resettableDelay"/>.
+        /// Gate to guard mutation of <see cref="_resettableDelay"/>.
         /// </summary>
         private readonly object _gate = new object();
 
-        private CancellationTokenSource? _cancellationTokenSource;
+        private readonly CancellationSeries _cancellationSeries = new();
         private ResettableDelay? _resettableDelay;
 
         public SourceGeneratedFileItemSource(SourceGeneratorItem parentGeneratorItem, Workspace workspace, IAsynchronousOperationListener asyncListener, IThreadingContext threadingContext)
         {
+            // Construction of BulkObservableCollection requires the main thread
```

Comment:
nit: consider moving the initialization into the ctor to make it more obvious

Reply:
💭 This feels like a change that someone would come along and move back.

Reply:
Oh yeah this confused me as well. @genlu's advice seems wise.

Reply:
➡️ Amended this commit to move the initialization to constructor

Reply:
Thanks, much cleaner!